falcon-regression/merlin-core/src/main/java/org/apache/falcon/regression/core/util/NativeInstanceUtil.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.falcon.regression.core.helpers.ColoHelper;
+import org.apache.falcon.resource.InstancesResult;
+import org.apache.hadoop.security.authentication.client.AuthenticationException;
+import org.apache.log4j.Logger;
+import org.apache.oozie.client.CoordinatorAction;
+import org.testng.Assert;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+/**
+ * Util functions related to native instance test.
+ * In case of entity scheduled using native scheduler, we cannot use oozie coordinator to retrieve relevant information
+ * This util retrieves instance related information using falcon APIs
+ */
+public final class NativeInstanceUtil {
+
+    public static final int INSTANCES_CREATED_TIMEOUT = OSUtil.IS_WINDOWS ? 20 : 10;
+    private static final Logger LOGGER = Logger.getLogger(NativeInstanceUtil.class);
+
+    private NativeInstanceUtil() {
+        throw new AssertionError("Instantiating utility class...");
+    }
+
+    /**
+     * Waits till instance of specific entity will be created during timeout.
+     * Timeout is common for most of usual test cases.
+     * Using status API to retrieve instance information.
+     *
+     * @param cluster     ColoHelper - colo on which API to be executed
+     * @param entity      definition of entity which describes job
+     * @param startTime   start time of instance
+     * @param endTime     end time of instance
+     */
+    public static void waitTillInstancesAreCreated(ColoHelper cluster, Entity entity, String startTime, String endTime)
+        throws InterruptedException, IOException, AuthenticationException,
+            URISyntaxException {
+        waitTillInstancesAreCreated(cluster, entity, startTime, endTime, INSTANCES_CREATED_TIMEOUT);
+    }
+
+    /**
+     * Waits till instances of specific job will be created during specific time.
+     * Use this method directly in unusual test cases where timeouts are different from trivial.
+     * In other cases use waitTillInstancesAreCreated(ColoHelper,Entity,String,String)
+     * Using status API to retrieve instance information.
+     *
+     * @param cluster     ColoHelper - colo on which API to be executed
+     * @param entity      definition of entity which describes job
+     * @param startTime   start time of instance
+     * @param endTime     end time of instance
+     * @param totalMinutesToWait     total time(in minutes) to wait for instance creation
+     */
+    public static void waitTillInstancesAreCreated(ColoHelper cluster, Entity entity, String startTime,
+        String endTime, int totalMinutesToWait)
+        throws URISyntaxException, AuthenticationException, InterruptedException, IOException {
+        String entityName = entity.getName();
+        EntityType type = entity.getEntityType();
+        String params = "?start=" + startTime;
+        params += (endTime.isEmpty() ? "" : "&end=" + endTime);
+        totalMinutesToWait = (totalMinutesToWait*60)/10;
```

Comment:
I think I get what you are trying to do. But this does not seem intuitive. Since you are overwriting totalMinutesToWait variable (totalMinutesToWait = (totalMinutesToWait*60)/10) and now, the name totalMinutesToWait does not actually represent the total minutes wait will happen. How about something like this - 

totalSecondsToWait = totalMinutesToWait * 60;
for (int sleepSeconds = 0; sleepSeconds < totalSecondsToWait; sleepSeconds = sleepSeconds+10) {
// Check instance
 TimeUtil.sleepSeconds(10);
}